dbms/src/Interpreters/SystemLog.cpp

```diff
@@ -70,6 +79,11 @@ void SystemLogs::shutdown()
         trace_log->shutdown();
     if (text_log)
         text_log->shutdown();
+    if (metric_log)
+    {
+        metric_log->stopCollectMetric();
```

Comment:
Non uniform... Ok, I understand that it because we don't use inheritance and virtual methods in SystemLog. No change needed.